src/client/src/rt-components/platform/adapters/glue/window.ts

```diff
@@ -0,0 +1,191 @@
+import { WindowConfig } from '../types'
+import { Glue42 } from 'tick42-glue'
+
+type BrowserWindowProps = WindowConfig
+type GDWindow = Glue42.Windows.GDWindow
+
+let listOfOpenedWindows: GDWindow[] = []
+
+/**
+ * Glue42 has the ability to manage windows.The registered methods here give the ability to demonstrate different
+ * functionalities with the opened windows (Activity, Blotter, Spot windows)
+ */
+export const registerWindowMethods = (glue: any) => {
+  if (location.href.indexOf('spot') >= 0) {
+    return
+  }
+
+  let isCollapsed: boolean = false
+  glue.agm.register('toggleCollapse', async () => {
+    for (const wnd of listOfOpenedWindows) {
+      isCollapsed ? await wnd.expand() : await wnd.collapse()
+    }
+    isCollapsed = !isCollapsed
+  })
+
+  glue.agm.register('stackAllWindows', () => {
+    listOfOpenedWindows.forEach(async (wnd: GDWindow, i: number) => {
+      if (i > 0) {
+        await wnd.detachTab({ relativeTo: listOfOpenedWindows[i - 1], relativeDirection: 'bottom' })
+      }
+    })
+  })
+
+  glue.agm.register('tabAllWindows', () => {
+    listOfOpenedWindows.forEach(async (wnd: GDWindow, i: number) => {
+      if (i > 0) {
+        await listOfOpenedWindows[0].attachTab(wnd, i)
+      }
+    })
+  })
+
+  glue.agm.register('openWorkspace', (args: { symbol: string }) => {
+    (window as any).glue42gd.canvas.openWorkspace('Reactive Trader Workspace', { context: args })
+  })
+
+  glue.windows.onWindowRemoved((removedWnd: GDWindow) => {
+    listOfOpenedWindows = listOfOpenedWindows.filter((wnd: GDWindow) => wnd.id !== removedWnd.id)
+    if ((window as any).glue) {
+      glue.agm.invoke('toggleHeaderButtons', { numberOfOpenedWindows: listOfOpenedWindows.length })
+    }
+  })
+}
+
+/**
+ * Opens a new glue window
+ * @param {BrowserWindowProps} config
+ * @param {() => void} onClose
+ * @return {Promise<any>}
+ */
+export const openGlueWindow = async (config: BrowserWindowProps, onClose?: () => void) => {
+  if (!(window as any).glue) {
+    return
+  }
+  const glue: any = (window as any).glue
+  const myWindow: GDWindow = glue.windows.my()
+  const { name, width, height, url } = config
+  const {
+    left,
+    top,
+    modifiedWidth,
+    modifiedHeight,
+    relativeTo,
+    relativeDirection,
+  } = calculatePosition(myWindow, width, height, url)
+  const fullUrl = `${location.href.slice(0, -1)}${url}`
+  const isTabWindow: boolean = url.indexOf('spot') >= 0
+
+  const win = await glue.windows.open(name, fullUrl, {
+    title: capitalizeFirstLetter(name.split(' ')[0]),
+    width: Math.round(modifiedWidth),
+    height: Math.round(modifiedHeight),
+    left: Math.round(left),
+    top: Math.round(top),
+    relativeTo,
+    relativeDirection,
+    allowCollapse: false,
+    mode: 'tab',
+    tabGroupId: isTabWindow ? 'reactiveTraderCloudSpot' : name,
+  })
+
+  if (url.indexOf('spot') >= 0) {
```

Comment:
it's worth pushing this check into a convenience function so you keep things dry.